src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs

```diff
@@ -215,6 +218,102 @@ public static long Flt2LngOvf(float val)
             return ThrowIntOvf();
         }
 
+#if ARM
+        private const string RuntimeLibrary = "[MRT]";
+
+        [RuntimeImport(RuntimeLibrary, "RhpIDiv")]
```

Comment:
All other RuntimeImports are marked with `[MethodImplAttribute(MethodImplOptions.InternalCall)]`. We should mark these as well for consistency.

Reply:
@jkotas When I did it I got the following error:
```
clang-3.9 -target arm-linux-gnueabi --sysroot=/share/CORERT/corert/tests/../cross/rootfs/armel -B/share/CORERT/corert/tests/../cross/rootfs/armel/usr/lib/gcc/armv7l-tizen-linux-gnueabi/6.2.1 -L/share/CORERT/corert/tests/../cross/rootfs/armel/usr/lib/gcc/armv7l-tizen-linux-gnueabi/6.2.1 DivConst.o -o DivConst /share/CORERT/corert/bin/Linux.armel.Debug/sdk/libbootstrapper.a /share/CORERT/corert/bin/Linux.armel.Debug/sdk/libRuntime.a /share/CORERT/corert/bin/Linux.armel.Debug/sdk/libSystem.Private.CoreLib.Native.a /share/CORERT/corert/bin/Linux.armel.Debug/framework/System.Native.a /share/CORERT/corert/bin/Linux.armel.Debug/framework/libSystem.Globalization.Native.a -g -Wl,-rpath,'$ORIGIN' -pthread -lstdc++ -ldl -lm -luuid -lrt
DivConst.o: In function `S_P_CoreLib_Internal_Runtime_CompilerHelpers_MathHelpers(int0_t, double)':
/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:285: undefined reference to `[MRT]'
DivConst.o: In function `S_P_CoreLib_Internal_Runtime_CompilerHelpers_MathHelpers__UMod':
/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:297: undefined reference to `[MRT]'
DivConst.o: In function `S_P_CoreLib_Internal_Runtime_CompilerHelpers_MathHelpers__UDiv':
/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:247: undefined reference to `[MRT]'
DivConst.o: In function `S_P_CoreLib_Internal_Runtime_CompilerHelpers_MathHelpers__LDiv':
/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:273: undefined reference to `[MRT]'
DivConst.o: In function `S_P_CoreLib_Internal_Runtime_CompilerHelpers_MathHelpers__ULMod':
/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:309: undefined reference to `[MRT]'
DivConst.o:/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:259: more undefined references to `[MRT]' follow

```

Reply:
This means that the `[MRT]` library name is somehow treated as an entry point name. Could you push the commit that adds `[MethodImplAttribute(MethodImplOptions.InternalCall)]` that you are getting this error with? It is the right change. We need to find out why is the library name treated as an entry point name.

This cannot ever work correctly without `[MethodImplAttribute(MethodImplOptions.InternalCall)]`. We have all handling of RuntimeImports behind IsInternalCall checks in the compiler, e.g. https://github.com/dotnet/corert/blob/2b40a707239a9974a63cf992ce6a6f8303f5b362/src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/RyuJitNodeFactory.cs#L23 .  And so the method won't get hooked up correctly with the C implementation without it.

Reply:
Actually, I understand what is going on: You should delete `EntryPoint=` part in the RuntimeImports to fix this.

When you say: `RuntimeImport(RuntimeLibrary, EntryPoint="RhpUDiv")`, it binds to `RuntimeImportAttribute(string entry)` attribute constructor. We should make the fields of RuntimeImportAttribute private to avoid this mistake.

Reply:
@jkotas  It's not working

Reply:
Could you please push the change with `EntryPoint=` part removed, and share the errors that you seeing with it?

Reply:
BTW: I have opened #4811 to make the RuntimeImport attribute less confusing.

Reply:
@jkotas 
```
clang-3.9 -target arm-linux-gnueabi --sysroot=/share/CORERT/corert/tests/../cross/rootfs/armel -B/share/CORERT/corert/tests/../cross/rootfs/armel/usr/lib/gcc/armv7l-tizen-linux-gnueabi/6.2.1 -L/share/CORERT/corert/tests/../cross/rootfs/armel/usr/lib/gcc/armv7l-tizen-linux-gnueabi/6.2.1 DivConst.o -o DivConst /share/CORERT/corert/bin/Linux.armel.Debug/sdk/libbootstrapper.a /share/CORERT/corert/bin/Linux.armel.Debug/sdk/libRuntime.a /share/CORERT/corert/bin/Linux.armel.Debug/sdk/libSystem.Private.CoreLib.Native.a /share/CORERT/corert/bin/Linux.armel.Debug/framework/System.Native.a /share/CORERT/corert/bin/Linux.armel.Debug/framework/libSystem.Globalization.Native.a -g -Wl,-rpath,'$ORIGIN' -pthread -lstdc++ -ldl -lm -luuid -lrt
DivConst.o: In function `S_P_CoreLib_Internal_Runtime_CompilerHelpers_MathHelpers(int0_t, double)':
/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:284: undefined reference to `[MRT]'
DivConst.o: In function `S_P_CoreLib_Internal_Runtime_CompilerHelpers_MathHelpers(int13_t, void)':
/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:234: undefined reference to `[MRT]'
DivConst.o: In function `S_P_CoreLib_Internal_Runtime_CompilerHelpers_MathHelpers__UMod':
/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:296: undefined reference to `[MRT]'
DivConst.o: In function `S_P_CoreLib_Internal_Runtime_CompilerHelpers_MathHelpers__UDiv':
/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:246: undefined reference to `[MRT]'
DivConst.o: In function `S_P_CoreLib_Internal_Runtime_CompilerHelpers_MathHelpers__LDiv':
/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:272: undefined reference to `[MRT]'
DivConst.o:/share/CORERT/corert/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs:308: more undefined references to `[MRT]' follow
clang: error: linker command failed with exit code 1 (use -v to see invocation)

```